src/core/public/overlays/flyout/flyout_service.tsx

```diff
@@ -119,9 +120,19 @@ export class FlyoutService {
 
         this.activeFlyout = flyout;
 
+        const onCloseFlyout = () => {
```

Comment:
@mshustov I've made another changes to the file in this commit. I needed to be able to have  a custom `onClose` handler that could prevent the flyout from closing.

Reply:
It doesn't look like `onClose`. `onClose` means a reaction to an action. Maybe it should be named` canClose` as well?

Reply:
Or `onClose` can accept `flyout` as an argument. Then a caller must call `flyout.close`:
```js
onClose = {(flyout) => if (condition) { flyout.close() } }
```

Reply:
Yes I'd prefer to stick to EUI `onClose` prop naming as it correspond to the "closing" event when clicking outside of the flyout. I will update with your second suggestion 👍